src/webapp/components/menu-button/MenuButton.tsx

```diff
@@ -0,0 +1,66 @@
+import i18n from "@eyeseetea/feedback-component/locales";
+import Button from "@material-ui/core/Button";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
+import KeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
+import KeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
+
+import { memo, useState, MouseEvent } from "react";
+import styled from "styled-components";
+
+type Item = {
+    label: string;
+    id: string;
+    handleClick: () => void;
+};
+
+type Props = {
+    label: string;
+    items: Item[];
+};
+
+export const MenuButton: React.FC<Props> = memo(({ label, items }) => {
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+    const onOpenMenu = (event: MouseEvent<HTMLButtonElement>) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+
+    return (
+        <Container>
+            <Button
+                aria-controls="simple-menu"
+                aria-haspopup="true"
+                variant="contained"
+                color="primary"
+                onClick={onOpenMenu}
+                endIcon={anchorEl ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
+            >
+                {i18n.t(label)}
+            </Button>
+            <Menu
+                id="simple-menu"
+                anchorEl={anchorEl}
+                keepMounted
+                open={Boolean(anchorEl)}
+                onClose={handleClose}
+            >
+                {items.map(item => (
+                    <MenuItem key={item.id} onClick={handleClose}>
+                        <Button variant="text" color="primary" onClick={item.handleClick}>
```

Comment:
I think this component should expose what `Item` was selected so the parent component can decide what to do instead passing a `handleClick` for each item:

```tsx
type Props = { label: string; items: Item[]; onItemSelected: (id: string) => void };

const handleClick = (item: Item) => {
  onItemSelected(item.id);
};

<Button variant="text" color="primary" onClick={() => handleClick(item)}>
```

Then in the parent component:

```tsx
const menuButtonItems = [
    {
        label: "Module 1",
        id: "module-1",
        handleClick: () => alert("Module 1"),
    },
    {
        label: "Module 2",
        id: "module-2",
        handleClick: () => alert("Module 2"),
    },
];

const onMenuItemSelected= (id: Id) => {
    const selectedItem= menuButtonItems.find(item=> item.id === id);
    // do something with the item
};

<MenuButton
    label={i18n.t("New Data Quality")}
    items={menuButtonItems}
    onItemSelected={onMenuItemSelected}
/>
```

does it make sense?